xrootd/authorization_test.go

```diff
@@ -295,9 +295,8 @@ func TestWriteOriginAuthFiles(t *testing.T) {
 
 	originAuthTester := func(server server_utils.XRootDServer, authStart string, authResult string) func(t *testing.T) {
 		return func(t *testing.T) {
-
+			defer viper.Reset()
```

Comment:
We don't want to just defer it, do we? It was at the top to ensure that there is a clean viper to begin with, just in case it wasn't reset in another test.

Reply:
If we do a `viper.Reset()` at the beginning of this, there's no way for me to set the viper variables that need to be configured for `EmitAuthfile` to get the right output.

I'll add an extra `viper.Reset()` at the beginning of the overall test, then we're guaranteed that the environment between these auth tests is clean.